Websockets crash due to missing property old_state

<!-- READ THIS FIRST:
- Make sure you run the latest version of the Android app
- Make sure you run the latest version of Home Assistant
- Make sure to check the Companion docs for troubleshooting and configuration: https://companion.home-assistant.io/
- Make sure the bug you found is not already reported, we love to put work in bugfixes instead of closing duplicate bug reports
  DO NOT DELETE ANY TEXT from this template! All requested information is important.
-->

**Home Assistant Android version:**
latest master
**Android version:**
12
**Phone model:**
Pixel 6 pro
**Home Assistant version:**
2021.11.4
**Last working Home Assistant release (if known):**
N/A
**Description of problem:**

Noticed a random crash from the app when I opened it and came to find the following traceback

**Traceback (if applicable, to get the logs you may refer to: https://companion.home-assistant.io/docs/troubleshooting/faqs/#android-crash-logs):**

```
2021-11-26 17:14:54.512 21774-21816/io.homeassistant.companion.android.debug E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-2
    Process: io.homeassistant.companion.android.debug, PID: 21774
    java.lang.IllegalArgumentException: Instantiation of [simple type, class io.homeassistant.companion.android.common.data.websocket.impl.entities.StateChangedEvent] value failed for JSON property old_state due to missing (therefore NULL) value for creator parameter oldState which is a non-nullable type
        at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: io.homeassistant.companion.android.common.data.websocket.impl.entities.EventResponse["data"]->io.homeassistant.companion.android.common.data.websocket.impl.entities.StateChangedEvent["old_state"])
        at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4394)
        at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4335)
        at io.homeassistant.companion.android.common.data.websocket.impl.WebSocketRepositoryImpl.handleEvent(WebSocketRepositoryImpl.kt:239)
        at io.homeassistant.companion.android.common.data.websocket.impl.WebSocketRepositoryImpl.access$handleEvent(WebSocketRepositoryImpl.kt:46)
        at io.homeassistant.companion.android.common.data.websocket.impl.WebSocketRepositoryImpl$onMessage$1.invokeSuspend(WebSocketRepositoryImpl.kt:284)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
     Caused by: com.fasterxml.jackson.module.kotlin.MissingKotlinParameterException: Instantiation of [simple type, class io.homeassistant.companion.android.common.data.websocket.impl.entities.StateChangedEvent] value failed for JSON property old_state due to missing (therefore NULL) value for creator parameter oldState which is a non-nullable type
        at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: io.homeassistant.companion.android.common.data.websocket.impl.entities.EventResponse["data"]->io.homeassistant.companion.android.common.data.websocket.impl.entities.StateChangedEvent["old_state"])
        at com.fasterxml.jackson.module.kotlin.KotlinValueInstantiator.createFromObjectWith(KotlinValueInstantiator.kt:116)
        at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:202)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:443)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1405)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:351)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:184)
        at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:542)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:563)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:438)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1405)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:351)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:184)
        at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4389)
        at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4335) 
        at io.homeassistant.companion.android.common.data.websocket.impl.WebSocketRepositoryImpl.handleEvent(WebSocketRepositoryImpl.kt:239) 
        at io.homeassistant.companion.android.common.data.websocket.impl.WebSocketRepositoryImpl.access$handleEvent(WebSocketRepositoryImpl.kt:46) 
        at io.homeassistant.companion.android.common.data.websocket.impl.WebSocketRepositoryImpl$onMessage$1.invokeSuspend(WebSocketRepositoryImpl.kt:284) 
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) 
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) 
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571) 
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750) 
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678) 
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665) 
```

**Screenshot of problem:**

**Additional information:**

--- a/common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt
+++ b/common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt
@@ -2,6 +2,7 @@ package io.homeassistant.companion.android.common.data.websocket.impl
 
 import android.util.Log
 import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.DeserializationFeature
 import com.fasterxml.jackson.databind.PropertyNamingStrategies
 import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
 import com.fasterxml.jackson.module.kotlin.readValue
@@ -55,6 +56,7 @@ class WebSocketRepositoryImpl @Inject constructor(
 
     private val ioScope = CoroutineScope(Dispatchers.IO + Job())
     private val mapper = jacksonObjectMapper()
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
         .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
     private val responseCallbackJobs = mutableMapOf<Long, CancellableContinuation<SocketResponse>>()
     private val id = AtomicLong(1)

